WrenchFu/gui.lua

```diff
@@ -11,52 +11,65 @@ Tweakable:
 ]]
 
 if not WrenchFu then WrenchFu = {} end
-WrenchFu.max_distance = 6
 
 
-function WrenchFu.open_gui_for(player_index, entity_name, position)
-    local reg = WrenchFu.handler_for(entity_name)
-
-    global.open_guis[player_index] = {
-        ["entity_name"] = entity_name,
-        ["position"] = position,
+function WrenchFu.open_gui_for(player_index, entity)
```

Comment:
The lines 24-28 handle the compatibility for mods using old version of interface.
Entity name and position only are used by WrenchFu tracking system, the client mod gets the entity (if it is registered via new interface).
The consealment of entity object seemed rather encumbering to me in the first place. In my [dronehome](url=https://forums.factorio.com/viewtopic.php?f=94&t=23226) mod the first thing I did after receiving the call from wrench was obtaining the entity as it was the single container for position, force and surface.
Other mods might have uses for entity in their further gui interaction processing.